### ISSUE TITLE
False positives when syntax is supported but not functionality

### ISSUE THREAD

**Issue by lohfu** [inspect-js/has-dynamic-import#2]
Node 10 (among others) supports the import syntax, but will throw a 'Not Supported' error when called. Currently this package returns true for versions that handle the syntax but does not actually support it. Your code appears to work partly because of the empty reject callback in `import.js`. Removing that callback will make it still return true but output an UnhandledPromiseRejectionWarning.

After fiddling around for a little bit, there seems to be no pretty way of doing this check synchronously. The only way i have found is using `util.inspect`, but seems a bit risky as the documentation says

> The output of util.inspect may change at any time and should not be depended upon programmatically.

EDIT: As long as `util.inspect` doesn't go bananas and starts outputting a something like "Not \<rejected\>", this should continue to work in all future versions of Node.

The following works (tested in node 0.8 - 16):

```js
// index.js
module.exports = function hasDynamicImport() {
	try {
		const promise = require('./import');

		if (util.inspect(promise).includes('<rejected>')) {
			// suppress UnhandledPromiseRejectionWarning
			promise.catch(() => {});
			return false;
		}
		return true;
	} catch (e) {
		return false;
	}
};

// import.js
module.exports = import('data:text/javascript,')
```

(Should maybe cache the result in the first call to prevent pointless reject callbacks to be added to the initial promise in case it is called a lot.)

Otherwise we have to do something like this:

```js
// index.js
module.exports = function hasDynamicImport() {
	try {
		return require('./import').then(() => true, () => false)
	catch (e) {
		return Promise.resolve(false)
	}
};

// somewhere else
if (await hasDynamicImport()) {
	// do stuff
}
```

**Comment by lohfu** [comment]
I created a PR, #3

**Comment by ljharb** [comment]
Fixed by #3.
